### PR TITLE
Defficiency medbay overhaul

### PR DESCRIPTION
**Pull request by Probe1**
### WIP

Just tossing this up here now so people know it's being actively worked on.  It will probably take me 3-5 days to finish.

**Please keep an open mind.  Everyone has their special interests.  Everyone wants [small part of Medbay I loved].  I will try to maintain as many QoL aspects as I can but there are some things that hold the entire departments quality hostage so we can have some creature comforts.  We are fixing _Medbay_
and not one little thing.**

### Image updated 1/26/2017

dreammaker_2017-01-26_08-21-14

TO DO:

Cameras
Lot of quality checks to make sure apcs are the right dir and such.
Proofchecking

Unfucking what I did to research when I lazily copy/pasted telesci and toxins

### CODE REVIEW

**Comment by Probe1** [comment]
That's fun.  Just editing step_x / y alone and establishing an entry ``` step_x = 0 ``` is enough to trigger travis.

**Comment by jakmak5** [comment]
looks pretty good. can we can some screenshots of the northern part of medbay?

**Comment by 9600bauds** [comment]
Fantastic placement of all Chemistry, Genetics and the Cloner

**Comment by Probe1** [comment]
There isn't too much to see @jakmak5 since the higher up you go the more raw it gets.

dreammaker_2017-01-16_03-15-14

**Comment by PJB3005** [comment]
Honestly we should reworks science too. That layout....

**Comment by Sailol** [comment]
I'd say that an airlock in telesci would be the one thing I'd add to this layout.

**Comment by Probe1** [comment]
dreammaker_2017-01-16_20-56-54

**Comment by Sprok0** [comment]
Adding an awful virology tumor to the outside of deff, shameful. Deff is the only map that has a proper internal viro and xenobio, it'd be nice to keep that.

**Comment by Probe1** [comment]
Having a completely sealed, constrained, windowless virology is one of the worst aspects of Medbay.

**Comment by Probe1** [comment]
>accidentally a step_y on a medical laptop
 >remove the step_y in notepad
 >clean the map
 >somehow it comes back

**Comment by kilozombie** [comment]
As a primary chemistry autist, that Chemistry fulfills my special needs. Thanks.

**Comment by Probe1** [comment]
image

@kilozombie this should be very close to the final for chemistry.

1.  Curbside desk
2.  Slot for players to stand at the fridge without blocking the hall
3.  Power square of chem dispensers
4. Trickypen Table Corner.
5.  TWO ENTRANCES
6.  Can see geneticists and develop a bromance of hating people that barge in.

**Comment by kilozombie** [comment]
Current deff chemistry has killed me a few times more than I want to admit, purely because there's one, cramped, oft-used entrance right next to another entrance so it's crowded 100% of the time, no other exit, three tables that block your passage but people can see you/shoot at you through. It works at roundstart, but as soon as there's ANY conflict in the center of the station (see: Chemistry), you're probably rarely ever going to be able to enter or exit.
One time I got a law 0 "do pranks" as a borg and put some very minor obstacles in front of the chemistry door during a virus and 90% of people ended up dying from the virus just because it was too cramped and chaotic to get in there and get out to distribute cures.
At least Box has an ultra-easy window to break and gain access, and the door is RARELY crowded with people trying to go places besides Chemistry.

I love this new setup. It seems like a lot of fun, you get to have a view of Genetics like you said, and it feels a lot more sophisticated and less clunky than any other Chemistry.

**Comment by AI-Is-Rogue** [comment]
>genetics not connected to science

aaaaaa

**Comment by Probe1** [comment]
I will miss having the random scientist walk in every third or fourth round.

I won't miss having a shitty Medbay that is crippled by trying to adhere to the 12 cardinal rules of "How its done on Box".

**Comment by Kurfursten** [comment]
Put the mancrowave against the wall south of the cryo pods, 2 wide is enough space and being able to quickly mousedrop someone into wake up machine without moving them far (running the risk of bumping someone and ending the drag) is paramount

It also might help your spacing issues if you just give paramedics telescience. It was tested out on Bus and was one of the only things people liked about that map. Scientists only use it to steal things, at least Paramedics have a legitimate use for it. Plus, science is not hurting for toys, while medbay is.

**Comment by 9600bauds** [comment]
What if you put a normal sleeper under Cryo and maybe make people realize the wake-up function exists and replaces the mancrowave PLUS reminds people that sleepers exist and work perfectly well on people who are not in crit :^)

**Comment by Kurfursten** [comment]
>three sleepers

Powercreep

**Comment by PJB3005** [comment]
That long stretch of hallway is looking awful honestly.

**Comment by kilozombie** [comment]
It could definitely do with some variation aesthetically, and also to spice up gameplay a bit. We only have a few long hallways on any given station and they're always public hallways, not ones for departments AFAIK.

**Comment by Probe1** [comment]
@PJB3005 Multi z  mapping when.

**Comment by Exxion** [comment]
When You Map It

We only have to change one (1) line of code to enable 3D ZAS, though said 3D ZAS would be very basic and to do it properly we'd need to do a fair bit more

**Comment by Probe1** [comment]
Interesting.

Anyway jokes aside if you have any brilliant ideas on how to make Deff not Deff I'm all for it.  I don't love a long hallway but at the same time it won't exactly hurt anyone. Completely reworking Research to accommodate Medbay will hurt me.

**Comment by Probe1** [comment]
I can't remember who ported layered pipes but I want to kill them.  Doesn't work on compile when mapped in, DD doesn't give any kind of warning or error that somethings been deleted, and I have no idea what's wrong with them.  But I will have to do a bunch of garbage mapping to make up for it.

dreamseeker_2017-01-20_04-15-13
dreammaker_2017-01-20_04-17-26

**Comment by Probe1** [comment]
New OP image

**Comment by PJB3005** [comment]
@Probe1

Comic made them from scratch.

I think you're just fucking up though because bus used them and it was fine.

**Comment by Probe1** [comment]
They're on the same layer.  When I get home tomorrow I'll look at Bus.  Thanks @PJB3005

**Comment by CptWad** [comment]
in that small empty section to the north in between viro and medbay throw in some beds and maybe a iv drip and a table with body bags on it

**Comment by ComicIronic** [comment]
multilayerpiping best code of my life

**Comment by SonixApache** [comment]
I greatly disagree with the movement of the reception desk and chemistry lab. 
There's also a lot of needless walls (rather than windows) that just block visibility everywhere.
I can't seem to find the second bioprinter near surgery.
Sec checkpoint is a nice touch, but maybe should have a window into the reception area.
Virology is better, red door should be a maint door however.

**Comment by 9600bauds** [comment]
I've already spoken plenty of the praises of this new layout so this time I'll omit that praise and ONLY point out the factual details, and this time, I did so in the most incredibly annoying to read and borderline unreadable way possible
cascsca
In red are the things that need fixin
In purple are my own personal requests that the world would not end if not fulfilled (some of them can be entirely chalked up to trying something different but I put them there anyways to make you :thinking:)

File: maps/defficiency.dmm [hunk 1]
```diff
@@ -12145,107 +12130,107 @@ aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
...
-aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaabaaaaaTaaaaabaaaaabaaaaaaaaaaabaaaaabaaaaaUaaaaabaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaXaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaVaabaaVaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
```
Comment (Zth--) [inline maps/defficiency.dmm]: are you sure about this line?

Comment (Probe1) [reply maps/defficiency.dmm]: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaah

Comment (9600bauds) [reply maps/defficiency.dmm]: aaaaaaaaaaaaaaaaaaaaaavaaaaaaaaaaaaaaaaaaaaaaaavaaaaaaaaaaaaaah

**Comment by Zth--** [comment]
Virology is a fish tank

**Comment by 9600bauds** [comment]
Another thing I noticed is that there are no rollerbeds or wheelchairs anywhere except the two puny ones in storage. It also lacks a fridge shower.

**Comment by Probe1** [comment]
I love you for giving feedback but it's on an image from last week that was off the cuff and is missing obvious "I hadn't done that yet" stuff.

**Comment by 9600bauds** [comment]
it's the most up to date I had ¯\\_(ツ)_/¯

**Comment by Probe1** [comment]
Basically all I need to do at this point is cameras, proofchecking again, and unfucking research.

dreammaker_2017-01-26_08-21-14

**Comment by Intigracy** [comment]
But the patient rooms 😢

**Comment by PJB3005** [comment]
Virology looks absolutely hideous, needs more maintenance too.

**Comment by Probe1** [comment]
What's wrong with it?

**Comment by PJB3005** [comment]
Just doesn't fit deff, especially the really long hallway.

**Comment by 9600bauds** [comment]
Put a bunch of microwaves in there and call it the microwave hallway.

**Comment by Probe1** [comment]
First I do the door room.  Then the microwave hallway.  

@PJB3005 I know this whole PR doesn't make you happy and I want to work with you.  Help me shape virology to be better.

Also the hallway feels less long in game thanks to the tile pattern and its low traffic priority.  It's absolutely long, and there's not much I can do about that without it being like Deffbay is now or eating maint or eating research.

**Comment by PJB3005** [comment]
@Probe1 maintenance.

Lots of maintenance.

**Comment by 9600bauds** [comment]
CHECKLIST
- [ ] Defibs + Defib chargers near Cryo and in Para
- [x] The missing prosthetics/organ printers near Surgery entrance
- [ ] Lack of rollerbeds and wheelchairs on the hallways
- [ ] PLEAS make the full bloodbags layer on top, just copypaste the table that is in meta's medbay it has nice pixel offsets and everything
- [ ] Lightswitches
- [ ] CMO needs rwalls instead of regular walls

**Comment by Probe1** [comment]
"Just copy and paste what you did on Meta"
CAN DO

**Comment by CptWad** [comment]
copy-pasting is never worth it

**Comment by 9600bauds** [comment]
We need to defib this PR

**Comment by CptWad** [comment]
probe dont be me and my failures
heres something to inspire you
image

**Comment by gbasood** [comment]
when

**Comment by Probe1** [comment]
This evening.

**Comment by Kurfursten** [comment]
If you finish this tonight, I'll finish Dr Acula by Saturday night

**Comment by Probe1** [comment]
The hangup has been

1. Work.  Really busy with real life.
2.  Admin/Server issues shitting the bed and making me focus on player issues.
3.  RE7 came out and my best friend and I played through it together.
4.  I don't know anything about Toxins and I'm really hesitant to fix it after I copied and pasted it.

But I'm not busy tonight with work, RE 7 is beaten and done, the servers running better, and I'll just copy bx or old deff and be done with it.

Also it feels like every person has a different opinion on how the map should be and it's all conflicting advice.  So I can't really do anything without making _someone_ upset and that really drains your will to create.

**Comment by Kurfursten** [comment]
This is the only advice you need to follow:

>Move the mancrowave closer to the pods

**Comment by Probe1** [comment]
Two tile corridor at cryo sounds like a nightmare.

**Comment by Kurfursten** [comment]
Put it to the left, shift the surgery door one over, move the freezer to where it is now

**Comment by SonixApache** [comment]
can we just code the mancrowave functionality into the sleepers so we can remove the damn thing already?

**Comment by Shadowmech88** [comment]
I'm pretty sure sleepers *already* have mancrowave functionality.

**Comment by SonixApache** [comment]
They wake you up, not sure if they heat you as well though.

**Comment by 9600bauds** [comment]
Cryo already heats you up when ejected (as it always did)

Even in the exceedingly rare scenario that someone has low bodytemp, you can dunk them in cryo and eject them

**Comment by Kurfursten** [comment]
This doesn't really matter because the sleepers are even further than the mancrowave is.

**Comment by 9600bauds** [comment]
F

**Comment by Probe1** [comment]
Conflicts aren't that bad. I'm old and wizened and can deal with them.  When you're new though.  Ho boy mapping conflicts when you're new.

**Comment by Kurfursten** [comment]
I will extend my offer. If you can finish this by Sunday noon, I'll finish Dr Acula by Monday noon

**Comment by SonixApache** [comment]
teach me your secrets probe

**Comment by Kurfursten** [comment]
Would even one person have thought that Dr Acula would be out before this?

**Comment by Kurfursten** [comment]
Arise
Arise PR
PR Arise

**Comment by ghost** [comment]
For what purpose

**Comment by PromethTK11** [comment]
For reviving it

**Comment by Riekland** [comment]
is probe even around anymore
did he go to the parlour in the sky as well

**Comment by PromethTK11** [comment]
rip

**Comment by Intigracy** [comment]
I haven't seen anything happening with this in four months, if you're still working on it let me know and I'll re-open it.

**Pull request closed without merging**